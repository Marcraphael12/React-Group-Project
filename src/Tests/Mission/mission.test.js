import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MissionsBody from '../../Pages/mission/missions_body';

describe('Display the missions list', () => {
  it('Renders correctly every missions from the API', () => {
    const mission = {
      id: 2,
      mission_name: 'one',
      description: 'a mission',
      joined: true,
    };
    const missions = TestRenderer.create(
      <Provider store={store}>
        <MissionsBody key={mission.id} missions={mission} />
      </Provider>,
    ).toJSON();
    expect(missions).toMatchSnapshot();
  });
});
