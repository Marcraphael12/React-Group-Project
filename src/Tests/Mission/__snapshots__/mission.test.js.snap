// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display the missions list Renders correctly every missions from the API 1`] = `
<div
  className="container"
>
  <table
    className="table table-bordered table-striped"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description 
        </th>
        <th>
          Status
        </th>
        <th>
          Join
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
